src/mqtt/token.h

```diff
@@ -250,7 +250,7 @@ class token : public virtual itoken
 	 * token.
 	 * @return int
 	 */
-	virtual int get_message_id() const { return int(tok_); }
+	virtual int get_message_id() const { return static_cast<int>(tok_); }
```

Comment:
In my opinion should be here an assertion here to ensure, at compile time, that tok_'s type (MQTTAsync_token) can be represented by an int:
`static_assert(sizeof(tok_) == sizeof(int))` 
otherwise we might produce narrowing bugs.